File: go/0238-product-of-array-except-self.go
func productExceptSelf(nums []int) []int {
    length := len(nums)
    prod := make([]int, length) 
    right := 1
    prod[0] = 1
    for i:= 1; i<length; i++ {
        prod[i] = prod[i-1]*nums[i-1]
    }
    for i:=length-1; i>=0; i-- {
        prod[i] = prod[i]*right
        right = right*nums[i]
    }
    return prod
}

File: python/0238_product_of_array_except_self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        pLeft = [0]*(len(nums))
        pRight = [0]*(len(nums))

        pLeft[0] = 1
        pRight[-1] = 1

        for i in range(1,len(nums)):
            pLeft[i] = pLeft[i-1]*nums[i-1]
        
        for i in range(len(nums)-2,-1,-1):
            pRight[i] = pRight[i+1]*nums[i+1]

        for i in range(0,len(nums)):
            pLeft[i] = pLeft[i]*pRight[i]
        
        return pLeft
